packages/remix-react/components.tsx

```diff
@@ -1309,3 +1310,18 @@ function useComposedRefs<RefValueType = any>(
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, refs);
 }
+
+let OutletContext = React.createContext<AppData>(null);
+
+export function Outlet<Data extends AppData = AppData>({ context }: { context?: Data }) {
```

Comment:
The context value should *not* be `AppData` because we don't want people using this for data they already have access to in `useMatches`.